Parse/Internal/PFDevice.m

```diff
@@ -27,14 +27,19 @@
     const char *charName = [name UTF8String];
 
     size_t size;
-    sysctlbyname(charName, NULL, &size, NULL, 0);
+    if (sysctlbyname(charName, NULL, &size, NULL, 0) < 0) {
+      return nil;
+    }
     char *answer = (char*)malloc(size);
 
     if (answer == NULL) {
         return nil;
     }
 
-    sysctlbyname(charName, answer, &size, NULL, 0);
+    if (sysctlbyname(charName, answer, &size, NULL, 0) < 0) {
+      free(answer);
```

Comment:
I also don't like the idea of having free in multiple places... I have a couple of thoughts on ways to clean this up...

1. Use a C99 VLA for answer instead of `malloc`.
2. Refactor to use a single exit point.

This situation right here is why most maintainers of C code are adamant about single exit points, as cleanup only happens in a single place :)